src/modules/27-platform/secrets/components/SecretReference/__test__/SecretReference.test.tsx

```diff
@@ -6,70 +6,291 @@
+          onSelect={noop}
+        />
       </TestWrapper>
     )
     await waitFor(() => getByText(container, 'entityReference.apply'))
-    expect(getByText(container, 'account')).toBeTruthy()
-    expect(container).toMatchSnapshot()
+
+    // Check if correct tabs are getting rendered
+    const projectTab = screen.getByText('projectLabel')
+    const orgTab = screen.getByText('orgLabel')
+    const accountTab = screen.getByText('account')
+    expect(projectTab).toBeInTheDocument()
+    expect(orgTab).toBeInTheDocument()
+    expect(accountTab).toBeInTheDocument()
+    // Check if secret from response is appearing on screen
+    const secretName = screen.getByText('text1')
+    expect(secretName).toBeInTheDocument()
   })
+
   test('render with spinner', async () => {
+    const listSecretsV2PromiseMock = jest.fn(() =>
+      Promise.resolve({
+        loading: true,
+        error: null,
+        data: null,
+        refetch: jest.fn().mockReturnValue(null)
+      })
+    )
+    ;(listSecretsV2Promise as jest.Mock).mockImplementation(listSecretsV2PromiseMock)
+
     const { container } = render(
       <TestWrapper>
-        <SecretReference accountIdentifier="dummy" mock={mockData as any} onSelect={noop} />
+        <SecretReference accountIdentifier="dummy" onSelect={noop} />
       </TestWrapper>
     )
     expect(getByText(container, 'account')).toBeTruthy()
-    expect(queryByAttribute('data-icon', container, /spinner/)).toBeTruthy()
-    expect(container).toMatchSnapshot()
+    expect(queryByAttribute('data-icon', container, /spinner/)).toBeInTheDocument()
+    const secretName = screen.queryByText('text1')
+    expect(secretName).not.toBeInTheDocument()
   })
+
   test('render with no data', async () => {
-    mockData.data.empty = true
-    mockData.data.content = []
+    const listSecretsV2PromiseMock = jest.fn(() =>
+      Promise.resolve({
+        loading: false,
+        error: null,
+        data: { data: { content: [] } },
+        refetch: jest.fn().mockReturnValue(null)
+      })
+    )
+    ;(listSecretsV2Promise as jest.Mock).mockImplementation(listSecretsV2PromiseMock)
+
     const { container } = render(
       <TestWrapper>
-        <SecretReference accountIdentifier="dummy" mock={mockData as any} onSelect={noop} />
+        <SecretReference accountIdentifier="dummy" onSelect={noop} />
       </TestWrapper>
     )
     await waitFor(() => getByText(container, 'entityReference.apply'))
-    expect(getByText(container, 'platform.secrets.secret.noSecretsFound')).toBeTruthy()
+    expect(getByText(container, 'platform.secrets.secret.noSecretsFound')).toBeInTheDocument()
     await waitFor(() => {
       // for SelectTypeDropdown (text/file)
       expect(getByText(container, 'platform.secrets.secret.labelSecretType')).toBeInTheDocument()
     })
-    expect(container).toMatchSnapshot()
+
+    const secretName = screen.queryByText('text1')
+    expect(secretName).not.toBeInTheDocument()
```

Comment:
No real benefit to extracting this to a variable, might as well inline it.
```suggestion
    expect(screen.queryByText('text1')).not.toBeInTheDocument()
```

Reply:
Makes code more readable to me. 😄 
But, I agree, no real benefit.